### PR TITLE
KServe modelmesh traffic engineering use cases

### PR DESCRIPTION
**Pull request by kalantar**

### CODE REVIEW

File: docs/tutorials/integrations/kserve-mm/blue-green.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      storageUri: s3://modelmesh-example-models/sklearn/mnist-svm.joblib
+EOF
+```
+
+??? note "About the primary `InferenceService`"
+    Naming the model with the suffix `-0` (and the candidate with the suffix `-1`) simplifies the rollout initialization. However, any names can be specified.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/blue-green.md:50]:
```suggestion
    Naming the model with the suffix `-0` (and the candidate with the suffix `-1`) simplifies the rollout initialization. However, any name can be specified.
```

File: docs/tutorials/integrations/kserve-mm/blue-green.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+```shell
+cat <<EOF | helm template traffic --repo https://iter8-tools.github.io/hub traffic-templates -f - | kubectl apply -f -
+templateName: initialize-rollout
+targetEnv: kserve-modelmesh
+trafficStrategy: blue-green
+modelName: wisdom
+EOF
+```
+
+The `initialize-rollout` template (with `trafficStrategy: blue-green`) configures the Istio service mesh to route all requests to the primary version of the model (`wisdom-0`). Further, it defines the routing policy that will be used by Iter8 when it observes changes in the models. By default, this routing policy splits inference requests 50-50 between the primary and candidate versions. For detailed configuration options, see the helm chart.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/blue-green.md:75]:
```suggestion
The `initialize-rollout` template (with `trafficStrategy: blue-green`) configures the Istio service mesh to route all requests to the primary version of the model (`wisdom-0`). Further, it defines the routing policy that will be used by Iter8 when it observes changes in the models. By default, this routing policy splits inference requests 50-50 between the primary and candidate versions. For detailed configuration options, see the Helm chart.
```

File: docs/tutorials/integrations/kserve-mm/canary.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
+---
+template: main.html
+---
+
+# Canary Rollout of a ML Model
+
+This tutorial shows how Iter8 can be used to implement a canary rollout of ML models hosted in a KServe modelmesh serving environment. In a canary rollout, inference requests that match a particular pattern, for example those that have a particular header, are directed to the candidate version of the model. The remaining requests go to the primary, or initial, version of the model. Iter8 enables a canary rollout by automatically configuring the network to distribute inference requests.
+
+After a one time initialization step, the end user merely deploys candidate models, evaluates them and either promotes deletes them. Iter8 automatically handles the underlying network configuration.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/canary.md:9]:
```suggestion
After a one time initialization step, the end user merely deploys candidate models, evaluates them, and either promotes or deletes them. Iter8 automatically handles the underlying network configuration.
```

File: docs/tutorials/integrations/kserve-mm/blue-green.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+      modelFormat:
+        name: sklearn
+      storageUri: s3://modelmesh-example-models/sklearn/mnist-svm.joblib
+EOF
+```
+
+??? note "About the primary `InferenceService`"
+    Naming the model with the suffix `-0` (and the candidate with the suffix `-1`) simplifies the rollout initialization. However, any names can be specified.
+    
+    The label `iter8.tools/watch: "true"` lets Iter8 know that it should pay attention to changes to this InferenceService.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/blue-green.md:52]:
```suggestion
    The label `iter8.tools/watch: "true"` lets Iter8 know that it should pay attention to changes to this `InferenceService`.
```

Most `InferenceService` seem to have the code highlight. I'm not sure if this is the only one missing it.

File: docs/tutorials/integrations/kserve-mm/canary.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+      modelFormat:
+        name: sklearn
+      storageUri: s3://modelmesh-example-models/sklearn/mnist-svm.joblib
+EOF
+```
+
+??? note "About the primary `InferenceService`"
+    Naming the model with the suffix `-0` (and the candidate with the suffix `-1`) simplifies the rollout initialization. However, any names can be specified.
+    
+    The label `iter8.tools/watch: "true"` lets Iter8 know that it should pay attention to changes to this InferenceService.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/canary.md:53]:
```suggestion
    The label `iter8.tools/watch: "true"` lets Iter8 know that it should pay attention to changes to this `InferenceService`.
```

File: docs/tutorials/integrations/kserve-mm/canary.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+```shell
+cat <<EOF | helm template traffic --repo https://iter8-tools.github.io/hub traffic-templates -f - | kubectl apply -f -
+templateName: initialize-rollout
+targetEnv: kserve-modelmesh
+trafficStrategy: canary
+modelName: wisdom
+EOF
+```
+
+The `initialize-rollout` template (with `trafficStrategy: canary`) configures the Istio service mesh to route all requests to the primary version of the model (`wisdom-0`). Further, it defines a routing policy that will be used by Iter8 when it observes changes in the models. By default, this routing policy sends inference requests with the header `traffic` set to the value `test` to the candidate version of the model and all remaining inference requests to the primary version of the model. For detailed configuration options, see the helm chart.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/canary.md:76]:
```suggestion
The `initialize-rollout` template (with `trafficStrategy: canary`) configures the Istio service mesh to route all requests to the primary version of the model (`wisdom-0`). Further, it defines a routing policy that will be used by Iter8 when it observes changes in the models. By default, this routing policy sends inference requests with the header `traffic` set to the value `test` to the candidate version of the model and all remaining inference requests to the primary version of the model. For detailed configuration options, see the Helm chart.
```

File: docs/tutorials/integrations/kserve-mm/mirror.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
+---
+template: main.html
+---
+
+# Mirrored Rollout of a ML Model
+
+This tutorial shows how Iter8 can be used to implement a mirrored rollout of ML models in a KServe modelmesh serving environment. In a mirrored rollout, all inference requests are sent to the primary version of the model. In addition, a portion of the requests are also sent to the candidate version of the model. The responses from the candidate version are ignored. Iter8 enables a mirrored rollout by automatically configuring the network to distribute inference requests.
+
+After a one time initialization step, the end user merely deploys candidate models, evaluates them and either promotes or deletes them. Optionally, the end user can modify the percentage of inference requests being sent to the candidate version of the model. Iter8 automatically handles the underlying network configuration.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/mirror.md:9]:
```suggestion
After a one time initialization step, the end user merely deploys candidate models, evaluates them, and either promotes or deletes them. Optionally, the end user can modify the percentage of inference requests being sent to the candidate version of the model. Iter8 automatically handles the underlying network configuration.
```

File: docs/tutorials/integrations/kserve-mm/mirror.md [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+```shell
+cat <<EOF | helm template traffic --repo https://iter8-tools.github.io/hub traffic-templates -f - | kubectl apply -f -
+templateName: initialize-rollout
+targetEnv: kserve-modelmesh
+trafficStrategy: mirror
+modelName: wisdom
+EOF
+```
+
+The `initialize-rollout` template ( with `trafficStrategy: mirror`) configures the Istio service mesh to route all requests to the primary model (`wisdom-0`). Further, it defines a routing policy that will be used by Iter8 when it observes changes in the models. By default, this routing policy sends all inference requests to the primary version of the model. It also sends them all of them (100 %) to the candidate version of the model. Responses from the candidate are ignored.
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve-mm/mirror.md:76]:
```suggestion
The `initialize-rollout` template ( with `trafficStrategy: mirror`) configures the Istio service mesh to route all requests to the primary model (`wisdom-0`). Further, it defines a routing policy that will be used by Iter8 when it observes changes in the models. By default, this routing policy sends all inference requests to the primary version of the model. It also sends them all of them (100%) to the candidate version of the model. Responses from the candidate are ignored.
```

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**